src/tableutils.js

```diff
@@ -253,6 +253,60 @@ export default class TableUtils extends Plugin {
 		} );
 	}
 
+	/**
+	 * Removes a row from the given `table`.
+	 *
+	 * This method properly re-calculate table geometry including `rowspan` attribute of any table cell that is overlapping removed row
+	 * and table headings values.
+	 *
+	 *		editor.plugins.get( 'TableUtils' ).removeRows( table, { at: 1, rows: 2 } );
+	 *
+	 * Assuming the table on the left, the above code will transform it to the table on the right:
+	 *
+	 *		row index
+	 *		    ┌───┬───┬───┐        `at` = 1        ┌───┬───┬───┐
+	 *		  0 │ a │ b │ c │        `rows` = 2      │ a │ b │ c │ 0
+	 *		    │   ├───┼───┤                        │   ├───┼───┤
+	 *		  1 │   │ d │ e │  <-- remove from here  │   │ i │ j │ 1
+	 *		    │   ├───┼───┤        will give:      ├───┼───┼───┤
+	 *		  2 │   │ g │ h │                        │ k │ l │ m │ 2
+	 *		    │   ├───┼───┤                        └───┴───┴───┘
+	 *		  3 │   │ i │ j │
+	 *		    ├───┼───┼───┤
+	 *		  4 │ k │ l │ m │
+	 *		    └───┴───┴───┘
+	 *
+	 * @param {module:engine/model/element~Element} table
+	 * @param {Object} options
+	 * @param {Number} options.at The row index at which the removing rows will start.
+	 * @param {Number} [options.rows=1] The number of rows to remove.
+	 */
+	removeRows( table, options ) {
+		const model = this.editor.model;
+		const first = options.at;
+		const rowsToRemove = options.rows || 1;
+
+		const last = first + rowsToRemove - 1;
+
+		model.change( writer => {
+			for ( let i = last; i >= first; i-- ) {
+				removeRow( table, i, writer );
```

Comment:
Not sure if related to this PR but (but it could be so worth checking):

1. Go to https://en.wikipedia.org/wiki/List_of_countries_and_dependencies_by_population.
2. Copy and paste the big countries table (243 rows in total).
3. Select some cell at the beginning of the table.
4. Scroll to the end (or near the end).
5. SHIFT+Click to create a huge selection across ~200 rows.
6. [Wait ages for the selection to create.](https://github.com/ckeditor/ckeditor5/issues/6568)
7. Delete rows.

**Expected**: All selected rows were deleted.

**Actual**: Around ~100 of them got deleted. The question is: why?

Reply:
OMG it is so funny :D 

:drum:

Reply:
Exectly 98 rows got deleted because...

Reply:
`[ 1, 2, 3, 4, 5, 10, 20, 30, 100, 1000, 2000 ].sort();`

is:

`[1, 10, 100, 1000, 2, 20, 2000, 3, 30, 4, 5]`

Reply:
I'll create an issue for this case.

Reply:
ckeditor/ckeditor5#6569